src/exchanges/dedupe.ts

```diff
@@ -0,0 +1,30 @@
+import { Observable, of } from 'rxjs';
+import { partition, merge, flatMap, tap } from 'rxjs/operators';
+import { Exchange, ExchangeResult, Operation } from '../types';
+
+export const dedupeExchange = (): Exchange => {
```

Comment:
- `operation` should not be shared, so `Map` can instead use `operation.key`
- `forward` is called too late (inside `flatMap`)
- maybe use `mergeMap` instead of the "obsolete" `flatMap` alias

Overall, we don't need to share the actual observables (see second point); Because it's an ongoing stream of operations/results we can take advantage of that and add the following two operations:

- On `ops$` do a continuous filter by active operations; can use `Set`
- On the results (`forward(filteredOps$)`) tap and remove operation keys from `Set`